internal/infrastructure/kubernetes/proxy_deployment.go

```diff
@@ -91,44 +91,18 @@ func (i *Infra) expectedProxyDeployment(infra *ir.Infra) (*appsv1.Deployment, er
 					SecurityContext:               deploymentConfig.Pod.SecurityContext,
 					Affinity:                      deploymentConfig.Pod.Affinity,
 					Tolerations:                   deploymentConfig.Pod.Tolerations,
-					Volumes: []corev1.Volume{
-						{
-							Name: "certs",
-							VolumeSource: corev1.VolumeSource{
-								Secret: &corev1.SecretVolumeSource{
-									SecretName: "envoy",
-								},
-							},
-						},
-						{
-							Name: "sds",
-							VolumeSource: corev1.VolumeSource{
-								ConfigMap: &corev1.ConfigMapVolumeSource{
-									LocalObjectReference: corev1.LocalObjectReference{
-										Name: expectedResourceHashedName(infra.Proxy.Name),
-									},
-									Items: []corev1.KeyToPath{
-										{
-											Key:  sdsCAFilename,
-											Path: sdsCAFilename,
-										},
-										{
-											Key:  sdsCertFilename,
-											Path: sdsCertFilename,
-										},
-									},
-									DefaultMode: pointer.Int32(int32(420)),
-									Optional:    pointer.Bool(false),
-								},
-							},
-						},
-					},
+					Volumes:                       expectedDeploymentVolumes(infra.Proxy.Name, deploymentConfig),
 				},
 			},
 		},
 	}
 
-	return deployment, nil
+	// apply merge patch to deployment
+	merged, err := deploymentConfig.ApplyMergePatch(deployment)
+	if err != nil {
+		return nil, fmt.Errorf("failed to apply merge patch to deployment: %w", err)
+	}
+	return merged, nil
```

Comment:
You can revert my suggestion here also to match upstream. Apologies, I missed one of the upstream PRs and thought you added this